formats/cbor/commonTest/src/kotlinx/serialization/cbor/CborStrictModeTest.kt

```diff
@@ -0,0 +1,20 @@
+package kotlinx.serialization.cbor
+
+import kotlinx.serialization.assertFailsWithMessage
+import kotlinx.serialization.decodeFromByteArray
+import kotlinx.serialization.HexConverter
+import kotlinx.serialization.DuplicateMapKeyException
+import kotlin.test.Test
+
+class CborStrictModeTest {
```

Comment:
TODO: Add a (passing) test for duplicate key detection when deserializing a data class. I have a test locally that fails -- I think I need to add ` visitKey` call to some inlined code somewhere, but I haven't found the appropriate callsite yet.

Reply:
I think I need to modify the serialization plugin's code generator in order to do this! Where does that code even live?

Reply:
Hmm. I had previously tried modifying `decodeElementIndex` and wasn't getting the results I expected. Maybe I'll take another crack at it.

Reply:
Oh! I see, the `decodeElementIndex` in `CborReader`. Got it.